File: naked_generator/bms/c/bms.c
#include <string.h>
#include <assert.h>
#include <stdio.h>

#include "bms.h"

/* bms_BMS_0 */
size_t serialize_bms_BMS_0(uint8_t* buffer, uint8_t crc_checksum, uint8_t pack_soc, uint16_t relay_state, uint16_t pack_inst_voltage, int16_t pack_current) {
    bms_BMS_0 bms_bms_0 = { crc_checksum, pack_soc, relay_state, pack_inst_voltage, pack_current };
	// assert(buf_len >= sizeof(bms_BMS_0));
	memcpy(buffer, &bms_bms_0, sizeof(bms_BMS_0));
    return sizeof(bms_BMS_0);
} 
size_t deserialize_bms_BMS_0(uint8_t* buffer, bms_BMS_0* bms_bms_0) {
	// assert(buf_len >= sizeof(bms_BMS_0));
	memcpy(bms_bms_0, buffer, sizeof(bms_BMS_0));
    return sizeof(bms_BMS_0);
}
/* bms_BMS_1 */
size_t serialize_bms_BMS_1(uint8_t* buffer, uint8_t crc_checksum, uint8_t low_temp, uint8_t high_temp, uint8_t pack_ccl, uint16_t pack_dcl) {
    bms_BMS_1 bms_bms_1 = { crc_checksum, low_temp, high_temp, pack_ccl, pack_dcl };
	// assert(buf_len >= sizeof(bms_BMS_1));
	memcpy(buffer, &bms_bms_1, sizeof(bms_BMS_1));
    return sizeof(bms_BMS_1);
} 
size_t deserialize_bms_BMS_1(uint8_t* buffer, bms_BMS_1* bms_bms_1) {
	// assert(buf_len >= sizeof(bms_BMS_1));
	memcpy(bms_bms_1, buffer, sizeof(bms_BMS_1));
    return sizeof(bms_BMS_1);
}

File: naked_generator/bms/js/bms.js
const byteify = require('byteify');

function serializeBmsBms0(data) {
    return Uint8Array.from([
        ...byteify.serializeUint8(data.crc_checksum),
        ...byteify.serializeUint8(data.pack_soc),
        ...byteify.serializeUint16(data.relay_state),
        ...byteify.serializeUint16(data.pack_inst_voltage),
        ...byteify.serializeInt16(data.pack_current),
    ]);
}
function deserializeBmsBms0(bytes) {
    return {
        crc_checksum: byteify.deserializeUint8(bytes.slice(0, 1)),
        pack_soc: byteify.deserializeUint8(bytes.slice(1, 2)),
        relay_state: byteify.deserializeUint16(bytes.slice(2, 4)),
        pack_inst_voltage: byteify.deserializeUint16(bytes.slice(4, 6)),
        pack_current: byteify.deserializeInt16(bytes.slice(6, 8)),
    }
}

function serializeBmsBms1(data) {
    return Uint8Array.from([
        ...byteify.serializeUint8(data.crc_checksum),
        ...byteify.serializeUint8(data.low_temp),
        ...byteify.serializeUint8(data.high_temp),
        ...byteify.serializeUint8(data.pack_ccl),
        ...byteify.serializeUint16(data.pack_dcl),
    ]);
}
function deserializeBmsBms1(bytes) {
    return {
        crc_checksum: byteify.deserializeUint8(bytes.slice(0, 1)),
        low_temp: byteify.deserializeUint8(bytes.slice(1, 2)),
        high_temp: byteify.deserializeUint8(bytes.slice(2, 3)),
        pack_ccl: byteify.deserializeUint8(bytes.slice(3, 4)),
        pack_dcl: byteify.deserializeUint16(bytes.slice(4, 6)),
    }
}
